src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -70,6 +70,7 @@ public function createShopListQueryBuilder(
         array $sorting = []
     ): QueryBuilder {
         $queryBuilder = $this->createQueryBuilder('o')
+            ->distinct()
```

Comment:
It's so simple when you see it, but couldn't figure it out when I was trying to solve this issue, good job 🎉